independent-projects/bootstrap/maven-resolver/src/main/java/io/quarkus/bootstrap/resolver/maven/NonBlockingModelResolutionTaskRunner.java

```diff
@@ -0,0 +1,84 @@
+package io.quarkus.bootstrap.resolver.maven;
+
+import java.util.Collection;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentLinkedDeque;
+import java.util.concurrent.Phaser;
+
+import org.jboss.logging.Logger;
+
+/**
+ * Non-blocking task runner implementation
```

Comment:
Maybe it also makes sense to document when it makes sense to use this implementation

Reply:
Basically, if the non-blocking failed, to check whether it due to concurrency issues :)
So that plus to see how much more efficient the non-blocking approach is.
I'll document it when I make it a proper option.

Reply:
👌